runtime-modules/forum/src/lib.rs

```diff
@@ -24,16 +25,11 @@ mod tests;
 /// Moderator ID alias for the actor of the system.
 pub type ModeratorId<T> = common::ActorId<T>;
 
+/// Forum user ID alias for the member of the system.
+pub type ForumUserId<T> = common::MemberId<T>;
+
 pub trait Trait: frame_system::Trait + pallet_timestamp::Trait + common::Trait {
     type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
-    type ForumUserId: Parameter
```

Comment:
This change here, switching this out with `common::MemberId<T>`, I suspect is emblematic for the sort of change we also discussed in the referendum? I think it is unfortunate to do this, as it makes modules, like the forum, coupled to unrelated code as a software artifact. Someone wanting to just use our forum module should not be dealing with this.

I do not know what problem this solves, but I worry now that this new `common` concept has become a locus of coupling for the entire runtime code base.

Reply:
Yes, `common::Trait` is a dependency for multiple modules. But we get this dependency only for modules that need membership binding and/or working group. Here, the forum user cannot be a non-member, because we check their rights to act in the forum using the working group. @iorveth  could correct me if I'm wrong.

We could refactor `common::Trait` and split it into two separate traits - MembershipBound (with MemberId defined) and ActorBound (with ActorId defined) to separate the need for membership from the need of the working group for pallets.

Also, we could see MemberId as a similar concept with the AccountId from the frame_system which is an unavoidable dependency. This is why this Trait was placed in the 'common' pallet. 

The alternative would be to implement a conversion code for each module that uses MemberId to/from the local identifier. It would also make the entire system not statically type-safe.

Reply:
I guess the trait separation seems like the best tradeoff, but lets not do it now, instead make an issue with tech-debt label, so then we can re-evaluate later.

Reply:
https://github.com/Joystream/joystream/issues/1996